### PR TITLE
fix(webserver): do not spawn webserver as a new process group

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch stops using `processLauncher` to launch web servers. Process Launcher
will spawn a new process group which is separate from test runner.

This might result in unexpected behavior, e.g. `kill -sigkill -<testrunnerpid>` will orphan web server process.

Instead, this patch simply spawns web server and sends `SIGTERM` to it
when ready.

### CODE REVIEW

**Comment by pavelfeldman** [comment]
Let's revert separately and then carefully look at the actual bug and the right fix.

File: packages/playwright-test/src/plugins/webServerPlugin.ts [hunk 1]
```diff
@@ -46,17 +47,15 @@ const debugWebServer = debug('pw:webserver');
 
 export class WebServerPlugin implements TestRunnerPlugin {
   private _isAvailable?: () => Promise<boolean>;
-  private _killProcess?: (presendSigtermBeforeSigkillTimeout?: number) => Promise<void>;
-  private _processExitedPromise!: Promise<any>;
+  private _processExitedPromise?: Promise<any>;
```
Comment (yury-s) [inline packages/playwright-test/src/plugins/webServerPlugin.ts:50]: Can it have a more specific type than `any`?

File: packages/playwright-test/src/plugins/webServerPlugin.ts [hunk 2]
```diff
@@ -66,22 +65,27 @@ export class WebServerPlugin implements TestRunnerPlugin {
...
   public async teardown() {
-    // Send SIGTERM and wait for it to gracefully close.
-    await this._killProcess?.(this._launchTerminateTimeout);
+    if (!this._childProcess || !this._childProcess.pid || this._childProcess.killed)
+      return;
+    try {
+      if (process.platform === 'win32')
+        execSync(`taskkill /pid ${this._childProcess.pid} /T /F /FI "MEMUSAGE gt 0"`, { stdio: 'ignore' });
+      else
+        process.kill(this._childProcess.pid, 'SIGTERM');
```
Comment (yury-s) [inline packages/playwright-test/src/plugins/webServerPlugin.ts:81]: Let's keep the comment about graceful termination ?

File: packages/playwright-test/src/plugins/webServerPlugin.ts [hunk 3]
```diff
@@ -92,47 +96,48 @@ export class WebServerPlugin implements TestRunnerPlugin {
...
-      attemptToGracefullyClose: async () => {},
-      log: () => {},
-      onExit: code => processExitedReject(new Error(code ? `Process from config.webServer was not able to start. Exit code: ${code}` : 'Process from config.webServer exited early.')),
-      tempDirectories: [],
     });
-    this._killProcess = kill;
+    this._processExitedPromise = new Promise((resolve, reject) => {
+      this._childProcess!.once('exit', code => {
+        if (code)
+          reject(new Error(`Process from config.webServer was not able to start. Exit code: ${code}`));
```
Comment (yury-s) [inline packages/playwright-test/src/plugins/webServerPlugin.ts:113]: I [believe](https://nodejs.org/api/child_process.html#event-exit) the exit code will be `null` when the process was killed by a signal. Should we check for both here?

File: tests/playwright-test/web-server.spec.ts [hunk 4]
```diff
@@ -602,27 +601,3 @@ test('should treat 3XX as available server', async ({ runInlineTest }, { workerI
...
-  const port = workerIndex + 10500;
-  const result = await runInlineTest({
-    'test.spec.ts': `
-      const { test } = pwt;
-      test('pass', async ({}) => {});
-    `,
-    'playwright.config.ts': `
-      module.exports = {
-        webServer: {
-          command: 'node ${JSON.stringify(SIMPLE_SERVER_THAT_IGNORES_SIGTERM_PATH)} ${port}',
```
Comment (yury-s) [inline tests/playwright-test/web-server.spec.ts:617]: Should we replace it with a test that checks graceful termination in response to sigterm or is it covered elsewhere already?

**Review by yury-s — APPROVED** [review]

File: tests/playwright-test/web-server.spec.ts (hunk 4 above)
Comment (aslushnikov) [reply tests/playwright-test/web-server.spec.ts]: All other tests check it!

**Pull request merged**